src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -2337,6 +2384,25 @@ export class ResultsViewPageStore {
         }
     }, []);
 
+    readonly nonSelectedMolecularProfiles = remoteData<MolecularProfile[]>({
+        await:()=>[this.molecularProfilesInStudies, this.selectedMolecularProfiles, this.molecularProfileIdToMolecularProfile],
```

Comment:
since you are trying to obtain the profiles, not the Ids, there is more efficient way to do this.  you can probably use _.difference directly on the arrays of profiles.   since the profile objects themselves are actually the same, they will have referential equality.    result = _.difference( allProfiles, selectedProfiles)